src/measurement-events/MeasurementEvent.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+
+/**
+ * Represents a single event that Shirshu tracks
+ *
+ * @class MeasurementEvent
+ */
+class MeasurementEvent {
+
+    /**
+     * The plugin that this event is associated with
+     *
+     * @var string
+     */
+    private $pluginName;
+
+    /**
+     * Event category e.g. ecommerce, engagement
+     *
+     * @var string
+     */
+    private $eventCategory;
+
+    /**
+     * Name of specific event
+     *
+     * @var string
+     */
+    private $eventAction;
+
+    /**
+     * CSS selector used to grab element that this event is tied to
+     *
+     * @var string
+     */
+    private $eventCssSelector;
+
+    /**
+     * Inner layer event to bind to in order to track the event
+     *
+     * @var string
+     */
+    private $onEvent;
+
+    /**
+     * Selector for element that contains the event to bind to when inner element is ready to be tracked.
+     *
+     * @var string
+     */
+    private $secondLayerSelector;
+
+    /**
+     * Second layer event to bind to that says when element is ready to be tracked
+     *
+     * @var string
+     */
+    private $secondLayerOnEvent;
+
+    /**
+     * MeasurementEvent constructor.
+     * @param $pluginName - string
+     * @param $category - string
+     * @param $action - string
+     * @param $selector - string
+     * @param $on - string
+     */
```

Comment:
Not sure if using a builder pattern would make this more readable

https://stackoverflow.com/questions/10961673/php-builder-pattern-without-inner-classes